libenv/unix_iface.c

```diff
@@ -1168,6 +1168,46 @@ JsonElement* GetProcFileInfo(EvalContext *ctx, const char* filename, const char*
 
 /*******************************************************************/
 
+static long GetInterfaceMTU(const char *iface)
+{
+    char sys_file_path[PATH_MAX] = {0};
+    int ret = snprintf(sys_file_path, PATH_MAX, "/sys/class/net/%s/mtu", iface);
```

Comment:
From man page:

> The snprintf() and vsnprintf() functions will write at most size-1 of the characters printed into the output string (the size'th character then gets the terminating '\0'); if the return value is greater than or equal to the size argument, the string was too short and some of the printed characters were discarded.  The output is always null-terminated, unless size is 0.

So as I understand it, you will (worst case) print `\0` to `sys_file_path[PATH_MAX]`, a buffer overflow of 1 byte.

According to:

https://pubs.opengroup.org/onlinepubs/009695399/basedefs/limits.h.html

PATH MAX includes the NUL byte:

> Maximum number of bytes in a pathname, including the terminating null character.

So I guess the correct thing is to do `PATH_MAX - 1` both here, and in the comparison below. For snprintf, both size param and return value should be treated as string length;

> These functions return the number of characters printed (not including the trailing '\0' used to end output to strings), except for snprintf() and vsnprintf(), which return the number of characters that would have been printed if the size were unlimited (again, not including the final '\0').

Reply:
Good catch, thanks!